ical_exporter.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import "io"
+
+const PRODID = "-//mlimaloureiro/golog"
+
+type ICalTaskExporter struct {
+	Version    string
+	CALSCALE   string
+	TimeFormat string // UTC time format string (layout string passed to time.Format)
+}
+
+// tryWriteStrings is a helper function used to write strings to a writer if an error has not previously occured
+// 	if an error occurs during the write the "error" argument will receive the error
+func tryWriteStrings(error *error, writer io.Writer, strings ...string) {
```

Comment:
this is a go anti pattern. You should deal with errors ASAP. 
Also new errors should be returned by the function and not mutating the previous one.

I would suggest you to take advantage of panic recover or use higher-order functions.

Reply:
My idea here was to make the code more readable and avoid repeating the pattern:
``` go
if err := writeString(writer, "BEGIN:VCALENDAR\n"); err != nil {
    return err
}

```
for every string that needed to be written to the stream. When writing many strings the code gets harder to read.

I ll change it to something like this:
``` go
if err := writeStrings(
	writer,
	"BEGIN:VEVENT\n",
	"SUMMARY:", startTask.getIdentifier(), "\n",
	"DTSTART:", startTime.UTC().Format(iCalTaskExporter.getTimeFormat()), "\n",
); err != nil {
	return err
}
```

Reply:
I understood what you try to accomplish that's why I made those suggestions.